Bugfix TraCI crash (#1776)
huawei-noah/SMARTS: CHANGELOG.md

- Added "SMARTS Performance Diagnostic" development tool for evaluating the simulation performance.
- Added a "All Simulation" button on the header of Envision and made small-windowed simulation(s) in the "All Simulations" page clickable to maximize.
- An env wrapper `Metrics` is introduced to compute agents' performance metrics.
- Extracted `TraciConn` to the SMARTS utilities as a simplified utility to help with connecting to `TraCI`.
### Deprecated
### Changed
- Minimum `SUMO` version allowed by `SumoTrafficSimulation` is now `1.10.0`.
### Removed
- Removed all of PyMarl contents, including related interface adapter, environments, and tests.
- Removed ray usage example.
- Moved ULTRA from `huawei-noah/SMARTS` to `smarts-project/smarts-project.rl` repository.
### Fixed
- Updated the RL example `racing` to use `smarts[camera_obs]==0.7.0rc0` and continuous flowing traffic scenario. Simplified the `racing` RL example folder structure.
- Envision "near realtime" mode bugfix
- SMARTS now aggressively attempts to connect to a SUMO process as long as the SUMO process remains alive.

> - SUMO traffic simulation `route_for_vehicle` works again.

Author: ```suggestion
- SUMO traffic simulation `route_for_vehicle` had syntax errors and now works again.
```

> - SUMO traffic simulation `route_for_vehicle` had syntax errors and now works again.

Author: ```suggestion
- SUMO traffic simulation `route_for_vehicle` had semantic errors and now works again.
```

> - SUMO is now supported up to version `1.15.0`. Versions `sumo>=1.12.0` are set to reset rather than reload because of errors with reset. 

Author: This was slightly inaccurate, the intended description is:

```suggestion
- SUMO is now supported up to version `1.15.0`. Versions of SUMO `1.13.0` and onward are forced to reset rather than reload because of errors with hot resetting versions starting with `1.13.0`. 
```
